### ISSUE TITLE
Uninitialized node causes the kube-router failure

### ISSUE THREAD

**Issue by kvaps** [cloudnativelabs/kube-router#846]
To reproduce that, try to create empty node:
```yaml
kubectl apply -f - <<EOT
apiVersion: v1
kind: Node
metadata:
  name: test
EOT
```

After that the kube-router pods will repeatedly dying on all the nodes with the following logs:
```
E0219 13:57:18.860221       1 bgp_peers.go:358] Error adding BGP policies: Failed to find a node IP: host IP unknown
E0219 13:57:18.900464       1 bgp_peers.go:162] Failed to add node <nil> as peer due to lookup <nil>: no such host
E0219 13:57:23.113111       1 ecmp_vip.go:100] Error adding BGP policies: Failed to find a node IP: host IP unknown
E0219 13:57:57.780511       1 network_routes_controller.go:264] Error synchronizing ipsets: Failed to find a node IP: host IP unknown
E0219 13:57:57.807345       1 network_routes_controller.go:292] Error adding BGP policies: Failed to find a node IP: host IP unknown
E0219 13:57:57.830794       1 bgp_peers.go:162] Failed to add node <nil> as peer due to lookup <nil>: no such host
E0219 13:57:57.830871       1 network_routes_controller.go:302] Error during periodic sync in network routing controller. Error: Failed to find a node IP: host IP unknown
E0219 13:57:57.830898       1 network_routes_controller.go:303] Skipping sending heartbeat from network routing controller as periodic sync failed.
E0219 13:58:00.607222       1 health_controller.go:118] Network Routing Controller heartbeat missed
E0219 13:58:02.607266       1 health_controller.go:118] Network Routing Controller heartbeat missed
E0219 13:58:03.607289       1 health_controller.go:118] Network Routing Controller heartbeat missed
E0219 13:58:06.607329       1 health_controller.go:118] Network Routing Controller heartbeat missed
E0219 13:58:07.607348       1 health_controller.go:118] Network Routing Controller heartbeat missed
time="2020-02-19T13:58:07Z" level=warning msg="received notification" Code=6 Communicated-Reason= Data="[]" Key=10.28.36.168 Subcode=2 Topic=Peer
time="2020-02-19T13:58:07Z" level=info msg="Peer Down" Key=10.28.36.168 Reason="notification-received code 6(cease) subcode 2(administrative shutdown)" State=BGP_FSM_ESTABLISHED Topic=Peer
I0219 13:58:07.608153       1 network_routes_controller.go:420] Cleaning up old routes if there are any
I0219 13:58:07.609337       1 network_routes_controller.go:434] Cleaning up if there is any existing tunnel interface for the node
E0219 13:58:07.609529       1 network_routes_controller.go:356] Failed to inject routes due to: no such process
time="2020-02-19T13:58:09Z" level=warning msg="received notification" Code=6 Communicated-Reason= Data="[]" Key=10.28.36.167 Subcode=2 Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Peer Down" Key=10.28.36.167 Reason="notification-received code 6(cease) subcode 2(administrative shutdown)" State=BGP_FSM_ESTABLISHED Topic=Peer
I0219 13:58:09.034007       1 network_routes_controller.go:420] Cleaning up old routes if there are any
I0219 13:58:09.035166       1 network_routes_controller.go:434] Cleaning up if there is any existing tunnel interface for the node
E0219 13:58:09.035354       1 network_routes_controller.go:356] Failed to inject routes due to: no such process
I0219 13:58:09.472002       1 kube-router.go:187] Shutting down the controllers
I0219 13:58:09.472241       1 health_controller.go:161] Shutting down health controller
I0219 13:58:09.472299       1 network_routes_controller.go:308] Shutting down network routes controller
I0219 13:58:09.472335       1 network_services_controller.go:359] Shutting down network services controller
I0219 13:58:09.472341       1 health_controller.go:178] Shutting down HealthController RunCheck
E0219 13:58:09.472413       1 health_controller.go:150] Health controller error: http: Server closed
time="2020-02-19T13:58:09Z" level=info msg="Administrative shutdown" Key=10.28.36.160 State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Administrative shutdown" Key=10.28.36.170 State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Administrative shutdown" Key=10.28.36.159 State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=warning msg="sent notification" Code=6 Communicated-Reason= Data="[]" Key=10.28.36.160 State=BGP_FSM_ESTABLISHED Subcode=2 Topic=Peer
time="2020-02-19T13:58:09Z" level=warning msg="sent notification" Code=6 Communicated-Reason= Data="[]" Key=10.28.36.170 State=BGP_FSM_ESTABLISHED Subcode=2 Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Administrative shutdown" Key=10.28.36.164 State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Administrative shutdown" Key=10.28.36.166 State=BGP_FSM_ESTABLISHED Topic=Peer
I0219 13:58:09.472816       1 network_policy_controller.go:166] Shutting down network policies controller
I0219 13:58:09.472300       1 metrics_controller.go:178] Shutting down metrics controller
time="2020-02-19T13:58:09Z" level=warning msg="sent notification" Code=6 Communicated-Reason= Data="[]" Key=10.28.36.166 State=BGP_FSM_ESTABLISHED Subcode=2 Topic=Peer
time="2020-02-19T13:58:09Z" level=warning msg="sent notification" Code=6 Communicated-Reason= Data="[]" Key=10.28.36.164 State=BGP_FSM_ESTABLISHED Subcode=2 Topic=Peer
time="2020-02-19T13:58:09Z" level=warning msg="sent notification" Code=6 Communicated-Reason= Data="[]" Key=10.28.36.159 State=BGP_FSM_ESTABLISHED Subcode=2 Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Administrative shutdown" Key=10.28.36.161 State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Peer Down" Key=10.28.36.160 Reason="notification-sent code 6(cease) subcode 2(administrative shutdown)" State=BGP_FSM_ESTABLISHED Topic=Peer
E0219 13:58:09.473041       1 metrics_controller.go:171] Metrics controller error: http: Server closed
time="2020-02-19T13:58:09Z" level=warning msg="sent notification" Code=6 Communicated-Reason= Data="[]" Key=10.28.36.161 State=BGP_FSM_ESTABLISHED Subcode=2 Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Administrative shutdown" Key=10.28.36.165 State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Administrative shutdown" Key=10.28.36.168 State=BGP_FSM_IDLE Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Peer Down" Key=10.28.36.164 Reason="notification-sent code 6(cease) subcode 2(administrative shutdown)" State=BGP_FSM_ESTABLISHED Topic=Peer
I0219 13:58:09.473350       1 network_routes_controller.go:420] Cleaning up old routes if there are any
time="2020-02-19T13:58:09Z" level=info msg="Peer Down" Key=10.28.36.166 Reason="notification-sent code 6(cease) subcode 2(administrative shutdown)" State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Peer Down" Key=10.28.36.170 Reason="notification-sent code 6(cease) subcode 2(administrative shutdown)" State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Administrative shutdown" Key=10.28.36.169 State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=warning msg="sent notification" Code=6 Communicated-Reason= Data="[]" Key=10.28.36.169 State=BGP_FSM_ESTABLISHED Subcode=2 Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Peer Down" Key=10.28.36.159 Reason="notification-sent code 6(cease) subcode 2(administrative shutdown)" State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Administrative shutdown" Key=10.28.36.167 State=BGP_FSM_IDLE Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Peer Down" Key=10.28.36.161 Reason="notification-sent code 6(cease) subcode 2(administrative shutdown)" State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=warning msg="sent notification" Code=6 Communicated-Reason= Data="[]" Key=10.28.36.165 State=BGP_FSM_ESTABLISHED Subcode=2 Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Peer Down" Key=10.28.36.165 Reason=read-failed State=BGP_FSM_ESTABLISHED Topic=Peer
time="2020-02-19T13:58:09Z" level=info msg="Peer Down" Key=10.28.36.169 Reason="notification-sent code 6(cease) subcode 2(administrative shutdown)" State=BGP_FSM_ESTABLISHED Topic=Peer
```

**Comment by murali-reddy** [comment]
thanks @kvaps for reporting the issue

I agree kube-router should not crash. But i am also wondering is it a acceptable node object? Should API server discard its creation?

**Comment by kvaps** [comment]
Yes, it's acceptable node object.
One of use casees for that might be to create nodes with all needed parameters (eg annotations and labels), and join kubelets later.

**Comment by aauren** [comment]
Closing as duplicate of #781

**Comment by aauren** [comment]
@kvaps After looking a bit more at this issue I can now see that while it is close to #781 it doesn't actually duplicate that issue. I've submitted a fix in #904
